numpy/core/setup_common.py

```diff
@@ -267,7 +267,7 @@ def check_long_double_representation(cmd):
         # try linking to support CC="gcc -flto" or icc -ipo
         # struct needs to be volatile so it isn't optimized away
         body = body.replace('struct', 'volatile struct')
-        body += "int main(void) { return 0; }\n"
+        body += "int main(void) { return foo.before[1]; }\n"
```

Comment:
This needs a comment explaining the reason. I'd also return `foo.before[0]`, the `1` just looks weird :)